### PR TITLE
Move edit functions and form to new component

### PR DESCRIPTION
**Pull request by dakota002**

### CODE REVIEW

**Review by lpsinger — CHANGES_REQUESTED** [review]
This PR does more than what I asked for in https://github.com/nasa-gcn/gcn.nasa.gov/pull/1680#discussion_r1436705317 because it introduces a new route, `_gcn.circulars.edit.$circularId`.

This PR should _only_ move the functions `useSubjectPlaceholder`, `useBodyPlaceholder`, `ToggleButton`, `useStateToggle`, `SyntaxExample`, `SyntaxReference`, and `CircularEditForm` into separate file(s).

File: app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+  const [body, setBody] = useState(defaultBody)
+  const bodyValid = bodyIsValid(body)
+  const [showKeywords, toggleShowKeywords] = useStateToggle(false)
+  const [showBodySyntax, toggleShowBodySyntax] = useStateToggle(false)
+  const [showPreview, setShowPreview] = useState(false)
+  const sending = Boolean(useNavigation().formData)
+  const valid = subjectValid && bodyValid
+  return (
+    <>
+      <h1>{circular ? 'Edit' : 'New'} GCN Circular</h1>
```
Comment (lpsinger) [inline app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx:132]: This PR still contains new code for history support. Would you please make sure that this PR is purely a refactoring of existing code?

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx (hunk 1 above)
Comment (dakota002) [reply app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx]: Should be all set now

File: app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx [hunk 2]
```diff
@@ -264,19 +246,8 @@ export function CircularEditForm({
...
-          {circular && (
-            <Button
-              disabled={sending || !valid}
-              type="submit"
-              name="intent"
-              value="distribute"
-            >
-              Update and Distribute
-            </Button>
-          )} */}
```
Comment (lpsinger) [inline app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx]: Remove commented out code.

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx (hunk 2 above)
Comment (dakota002) [reply app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx]: That should have been removed in the last commit

File: app/routes/_gcn.circulars.new.tsx [hunk 3]
```diff
@@ -69,224 +47,30 @@ export async function loader({ request }: LoaderFunctionArgs) {
...
-  const [body, setBody] = useState(defaultBody)
-  const bodyValid = bodyIsValid(body)
-  const [showKeywords, toggleShowKeywords] = useStateToggle(false)
-  const [showBodySyntax, toggleShowBodySyntax] = useStateToggle(false)
-  const [showPreview, setShowPreview] = useState(false)
-  const sending = Boolean(useNavigation().formData)
-  const valid = subjectValid && bodyValid
-
   return (
     <AstroDataContext.Provider value={{ rel: 'noopener', target: '_blank' }}>
```
Comment (lpsinger) [inline app/routes/_gcn.circulars.new.tsx:72]: Please move this up into the form component.

File: app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+          <InputPrefix className="wide-input-prefix">From</InputPrefix>
+          <span className="padding-1">{formattedContributor} </span>
+          <Link
+            to="/user"
+            title="Adjust how your name and affiliation appear in new GCN Circulars"
+          >
+            <Button unstyled type="button">
+              <Icon.Edit role="presentation" /> Edit
+            </Button>
+          </Link>
```
Comment (lpsinger) [inline app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx:141]:
```suggestion
          <span className="padding-1">{formattedContributor}{' '}
            <Link
              to="/user"
              title="Adjust how your name and affiliation appear in new GCN Circulars"
            >
              <Button unstyled type="button">
                <Icon.Edit role="presentation" /> Edit
              </Button>
            </Link>
        </span>
```

File: app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+          <Link
+            to={`/circulars${searchString}`}
+            className="usa-button usa-button--outline"
+          >
+            Back
+          </Link>
+          <Button
+            disabled={sending || !valid}
+            type="submit"
+            name="intent"
```
Comment (lpsinger) [inline app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx:246]: Not part of refactoring:

```suggestion
```

File: app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+  defaultBody,
+  defaultSubject,
+  searchString,
+  isAuthorized,
+}: {
+  formattedContributor: string
+  defaultBody: string
+  defaultSubject: string
+  searchString: string
+  isAuthorized: boolean
```
Comment (lpsinger) [inline app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx:110]: Unused:

```suggestion
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

File: app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+            </Link>
+          </span>
+        </InputGroup>
+        <InputGroup
+          className={classnames('maxw-full', {
+            'usa-input--error': subjectValid === false,
+            'usa-input--success': subjectValid,
+          })}
+        >
+          <InputPrefix>Subject</InputPrefix>
```
Comment (lpsinger) [inline app/routes/_gcn.circulars.edit.$circularId/CircularEditForm.tsx:149]:
```suggestion
          <InputPrefix className="wide-input-prefix">Subject</InputPrefix>
```

**Review by lpsinger — CHANGES_REQUESTED** [review]

**Review by lpsinger — APPROVED** [review]

**Pull request merged**